packages/runner-ct/src/iframe/iframes.tsx

```diff
@@ -22,134 +21,133 @@ export function getSpecUrl ({ namespace, spec }, prefix = '') {
-            transform: `scale(${screenshotting ? 1 : scale})`,
-          }}
+export const Iframes: React.FC<IFramesProps> = namedObserver('Iframes', (props) => {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [autIframe] = useState(new AutIframe(props.config))
+  const [disposers, setDisposers] = useState<IReactionDisposer[]>([])
+
+  const [iframeModel] = useState(
+    new IframeModel({
+      state: props.state,
+      restoreDom: autIframe.restoreDom,
+      highlightEl: autIframe.highlightEl,
+      detachDom: autIframe.detachDom,
+      snapshotControls: (snapshotProps) => (
+        <SnapshotControls
+          eventManager={props.eventManager}
+          snapshotProps={snapshotProps}
+          state={props.state}
+          onToggleHighlights={_toggleSnapshotHighlights}
+          onStateChange={_changeSnapshotState}
         />
-        <ScriptError error={scriptError} />
-        <div className='cover' />
-      </div>
-    )
-  }
-
-  componentDidMount () {
-    const config = this.props.config
-
-    this.autIframe = new AutIframe(config)
+      ),
+    }),
+  )
 
-    this.props.eventManager.on('visit:failed', this.autIframe.showVisitFailure)
-    this.props.eventManager.on('before:screenshot', this.autIframe.beforeScreenshot)
-    this.props.eventManager.on('after:screenshot', this.autIframe.afterScreenshot)
-    this.props.eventManager.on('script:error', this._setScriptError)
+  useEffect(() => {
+    props.eventManager.on('visit:failed', autIframe.showVisitFailure)
+    props.eventManager.on('before:screenshot', autIframe.beforeScreenshot)
+    props.eventManager.on('after:screenshot', autIframe.afterScreenshot)
+    props.eventManager.on('script:error', _setScriptError)
 
     // TODO: need to take headless mode into account
     // may need to not display reporter if more than 200 tests
-    this.props.eventManager.on('restart', () => {
-      this._run(this.props.state.spec, config)
+    props.eventManager.on('restart', () => {
+      _run(props.state.spec, props.config)
     })
 
-    this.props.eventManager.on('print:selector:elements:to:console', this._printSelectorElementsToConsole)
+    props.eventManager.on('print:selector:elements:to:console', autIframe.printSelectorElementsToConsole)
+
+    props.eventManager.start(props.config)
+
+    setDisposers([
+      autorun(() => {
+        autIframe.toggleSelectorPlayground(selectorPlaygroundModel.isEnabled)
+      }),
+      autorun(() => {
+        autIframe.toggleSelectorHighlight(selectorPlaygroundModel.isShowingHighlight)
+      }),
+      autorun(() => {
+        if (props.state.spec) {
+          _run(props.state.spec, props.config)
+        }
+      }),
+    ])
+
+    iframeModel.listen()
+
+    return () => {
+      props.eventManager.notifyRunningSpec(null)
+      props.eventManager.stop()
+      disposers.forEach((dispose) => {
+        dispose()
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I want this to be like `componentDidMount` (that how's the code worked, pre-refactor). My feeling is passing all the require args in the array is not going to do this, though. You can remove the eslint disable and run `yarn lint` and see what I mean.

Reply:
Yes. You have it written correctly. The ESLint rule doesn't cover a standard `componentDidMount` scenario.